### ISSUE TITLE
data for training

### ISSUE THREAD

**Issue by Watson52** [autonomousvision/transfuser#62]
Hi
When I trying to generate the data with different scenerios, I have some questions about the dataset.

1. I found the 14_weathers_minimal_data is divide by routes in differenct Town and one route includes kinds of scenarios. Could I divide the dataset by scenario? I mean that there is just one kind of scenario occur in one route. Does it will affect the result of training?
2. There are 3 types of routes: long/short/tiny. I would like to know how the training result will happen without 'long' routes. Or even just have 'tiny' routes, will it make a different to the result?

Looking forward to your help.

**Comment by ap229997** [comment]
1. It should not affect training as long as your data contains sufficient diversity of scenarios & routes.
2. The default setting in the paper uses short & tiny routes for training. If you use just 'tiny' routes, the agent will be very conservative (it'll stop a lot). If you use 'long' routes as well then the data distribution gets very skewed towards 'follow lane' situations and the agent may not be able to learn how to react in non-follow lane situations.

**Comment by Watson52** [comment]
> 1. It should not affect training as long as your data contains sufficient diversity of scenarios & routes.
> 2. The default setting in the paper uses short & tiny routes for training. If you use just 'tiny' routes, the agent will be very conservative (it'll stop a lot). If you use 'long' routes as well then the data distribution gets very skewed towards 'follow lane' situations and the agent may not be able to learn how to react in non-follow lane situations.

Thanks! Your explain help me a lot. 
btw, I want to design complex scenarios and use different/separated expert to generate data. Will I success to obtain an agent with experts' abilities? Or it depends on the specific situation to convergence.

**Comment by Watson52** [comment]
> 1. It should not affect training as long as your data contains sufficient diversity of scenarios & routes.
> 2. The default setting in the paper uses short & tiny routes for training. If you use just 'tiny' routes, the agent will be very conservative (it'll stop a lot). If you use 'long' routes as well then the data distribution gets very skewed towards 'follow lane' situations and the agent may not be able to learn how to react in non-follow lane situations.

Thanks! Your explain help me a lot. 
btw, I want to design complex scenarios and use different/separated expert to generate data. Will I success to obtain an agent with experts' abilities? Or it depends on the specific situation to convergence.

**Comment by Watson52** [comment]
Thank you for your patience.

**Comment by Watson52** [comment]
Hi, I try to generate data using the AutoPilot and I found the setting of lidar sensor only include the carla.Transform. 

1. Does it defaultly  use `channels=64` corresponding to Velodyne 64? Do I need to set other parameters such as `points_per_second`?
2. Why does the value of `yaw `set to `-90`?
https://github.com/autonomousvision/transfuser/blob/9d5ebf01f7b5f6e5c02ebe0cba423119c465b388/leaderboard/team_code/base_agent.py#L169-L174
Thank you in advance.

**Comment by ap229997** [comment]
The default settings are:
https://github.com/autonomousvision/transfuser/blob/9d5ebf01f7b5f6e5c02ebe0cba423119c465b388/leaderboard/leaderboard/autoagents/agent_wrapper.py#L149-L164

If you don't set yaw, you would need to rotate the coordinate axes to align them appropriately with the convention used in our code (see #38).

**Comment by Watson52** [comment]
Thanks! It helps me a lot. btw, where is the `AgentWrapper `called?

**Comment by ap229997** [comment]
https://github.com/autonomousvision/transfuser/blob/2af3e0226a3edc45fd5ac3a4545cc790cf04a73f/leaderboard/leaderboard/scenarios/scenario_manager.py#L105

**Comment by Watson52** [comment]
Thank you so much :)

**Comment by Watson52** [comment]
Hi, sorry to bother you again. I want to change the location of the lidar both for expert and agent, I just need to change the `z` axis coordinate values such as `-2` in the data processing. Is it right? Since I found that the lidar's blind spots are reduced through fine-tuning.
![1](https://user-images.githubusercontent.com/42103990/172527979-e9234688-77b9-479f-982d-f1f2ecf0579f.png)

**Comment by ap229997** [comment]
The ideal way is to change the location in the sensor configuration of the autopilot and generate data again.
